addons/openebs/1.12.0/install.sh

```diff
@@ -256,3 +259,149 @@ function openebs_cstor_pool_count() {
     local actual=$(kubectl -n "$OPENEBS_NAMESPACE" get cstorpools -l openebs.io/storage-pool-claim=cstor-disk --no-headers 2>/dev/null | wc -l)
     [ $actual -ge $target ]
 }
+
+function openebs_upgrade() {
+    # TODO: handle namespace spec changes at the upgrade time
+    if [ "$OPENEBS_CSTOR" = "0" ]; then
+        # upgrades only required for cStor 
+        return 0
+    fi
+
+    # RE: https://github.com/openebs/openebs/tree/master/k8s/upgrades/1.x.0-1.12.x
+    # Upgrading a minor version of 1.x.x requires two jobs to update pools a volumes for cStor configurations.
+    local runningVer=$(kubectl -n $OPENEBS_NAMESPACE get deploy openebs-provisioner -o jsonpath='{.metadata.labels.openebs\.io/version}')
+
+    local semVerRunningList=( ${runningVer//./ } )
+    local semVerInstallList=( ${OPENEBS_VERSION//./ } )
+    
+    if [[ ${semVerInstallList[0]} -gt 1 ]]; then
+        bail "Only upgrades upto OpenEBS 1.12.0 are tested and supported."
+    fi
+
+    logSubstep "Runnig upgrade checks..."
+    if [[ ${semVerInstallList[1]} -gt ${semVerRunningList[1]} || openebs_check_pools ]]; then
+        logSubstep "As part of OpenEBS upgrade pools and volumes need to be upgraded. At least some of the pools and(or) volumes require an update to match OpenEBS controll plane."
+        logFail "It is highly recommended to schedule a downtime for the application using the OpenEBS PV while performing this upgrade. Also, make sure you have taken a backup of the data before starting the below upgrade procedure."
+        printf "Continue? "
+        if ! confirmN " "; then
+            bail "Will not upgrade OpenEBS. Modify your spec and re-run instllation."
```

Comment:
why would they need to modify their spec?

Reply:
If a user decided to pospone upgrade. The running spec would have 1.12.0, in order to continue running 1.6.0 they would need to set it in the spec.

Reply:
I replaced the wording with "OpenEBS upgrade is aborted."